pki/certs.go

```diff
@@ -103,17 +103,18 @@ type CertificatePair struct {
 // to a log. The certificate's subject common name will be a random subdomain
 // based on the certificate serial under the `testCertDomain` domain.
 //
-// If windowStart is nil the certificate NotBefore will be set to the current
-// time based on the provided clock. If windowStart is not nil then the
-// certificate NotBefore will be set to the windowStart plus one day.
-
-// If windowEnd is nil the certificate NotAfter will be set to 90 days after the
-// current time based on the provided clock. If windowEnd is not nil then the
-// certificate NotAfter will be set to the windowEnd minus one day.
+// If windowEnd is nil, issue a 90 day certificate with validity starting today.
+// If windowEnd is in the future, issue a certificate with validity starting today
+// and ending just before windowEnd.
+//
+// If windowEnd is in the past, issue a 90 day certificate that expires
+// just before windowEnd. This allows monitoring of shards that are expired,
+// but not yet frozen.
 //
 // This function creates certificates that will be submitted to public logs and
 // so while they are not issued by a trusted root  we try to avoid cablint
 // errors to avoid requiring log monitors special-case our submissions.
+// In some cases we may issue certificates valid for longer than 825 days.
```

Comment:
Are we certain that the Trillian folks won't accept a PR that would allow future dated certificates? Is there an issue with the upstream to discuss? Having certificates with lifetimes that stretch across every shard is a bit clumsy and I'm worried we're picking this fix without having considered others first.